public/js/actions/sources.js

```diff
@@ -184,6 +184,25 @@ function togglePrettyPrint(id) {
   };
 }
 
+function loadOriginalSourceText(source, getState, dispatch) {
+    return Task.spawn(function* () {
+      const generatedSource = getSource(
+        getState(),
+        getGeneratedSourceId(source)
+      );
+
+      const generatedSourceText = yield dispatch(
+        loadSourceText(generatedSource.toJS())
+      );
+
+      return getOriginalSource(
+        source,
+        generatedSource.toJS(),
+        generatedSourceText
+      );
+    })
+}
```

Comment:
you might be asking why the sources action is now loading the original source instead of the source-map utility. The reason is that, loading the original source requires `getState` and `dispatch`, and i'm trying to keep these dependencies outside of the source-map utility.